pandas/core/generic.py

```diff
@@ -2458,8 +2458,12 @@ def to_msgpack(self, path_or_buf=None, encoding='utf-8', **kwargs):
         return packers.to_msgpack(path_or_buf, self, encoding=encoding,
                                   **kwargs)
 
-    def to_sql(self, name, con, schema=None, if_exists='fail', index=True,
-               index_label=None, chunksize=None, dtype=None, method=None):
+    def to_sql(self, name: str, con,
+               schema: Optional[str] = None, if_exists: str = 'fail',
```

Comment:
Shouldn't need the Optional for these; mypy should do this implicitly

Reply:
`Optional` is used throughout `typing` and `mypy` documentation ([here](https://docs.python.org/3/library/typing.html#typing.Optional) and [here](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#functions)). Reading one way of doing this in `typing`/`mypy` docs and another way in `pandas` code would confuse me. Let me know if you considered this already and I'll remove them with the next commit.

Reply:
Yea we are still pretty immature w.r.t. annotations so you will see some inconsistency. Since this doesn't add anything I'd prefer the shorter `str` to `Optional[str]` from a readability perspective.